miden-crypto/src/dsa/rpo_falcon512/signature.rs

```diff
@@ -33,19 +33,59 @@ use crate::Word;
 /// Here h is a polynomial representing the public key and pk is its digest using the Rpo256 hash
 /// function. c is a polynomial that is the hash-to-point of the message being signed.
 ///
-/// The polynomial h is serialized as:
-/// 1. 1 byte representing the log2(512) i.e., 9.
-/// 2. 896 bytes for the public key itself.
+/// Another differentiating point from the reference implementation is the determinism in its
+/// signing process. The approach used to achieve this is the one proposed in [1]. The main
+/// challenge in making the signing procedure deterministic is ensuring that the same secret key
+/// is never used to produce two inequivalent signatures for the same `c`. For a precise definition
+/// of equivalence of signatures see [1]. The reference implementation uses a random nonce per
+/// signature in order to make sure that, with overwhelming probability, no two c-s will ever
+/// repeat and this non-repetition turns out to be enough to make the security proof of
+/// the underlying construction go through in the random-oracle model.
+/// Making the signing process deterministic means that we cannot rely on the above use of nonce
+/// in the hash-to-point algorithm, i.e., the hash-to-point algorithm is deterministic. It also
+/// means that we have to derandomize the trapdoor sampling process and use the entropy in
+/// the secret key, together with the message, as the seed of a CPRNG. This is exactly the approach
+/// taken in [2] but, as explained at length in [1], this is not enough. The reason for this
+/// is that the sampling process during signature generation must be ensured to be consistent
+/// across the entire computing stack i.e., hardware, compiler, OS, sampler implementations ...
+/// This is made even more difficult by the extensive use of floating-point arithmetic by
```

Comment:
nit: maybe add a new line above this line too.

Reply:
done